specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json

```diff
@@ -0,0 +1,1121 @@
+              "description": "Recommended command to mount the AML file system",
+              "readOnly": true,
+              "type": "string"
+            },
+            "lustreVersion": {
+              "type": "string",
+              "description": "The version of Lustre running in the AML file system",
+              "readOnly": true
+            },
+            "throughputProvisionedMBps": {
+              "type": "integer",
+              "format": "int32",
+              "description": "Throughput provisioned in MB per sec, calculated as storageCapacityTiB * per-unit storage throughput",
+              "readOnly": true
+            },
+            "encryptionSettings": {
+              "$ref": "#/definitions/AmlFilesystemEncryptionSettings",
+              "description": "Specifies encryption settings of the AML file system."
+            },
+            "maintenanceWindow": {
+              "description": "Start time of a 30-minute weekly maintenance window.",
+              "type": "object",
+              "properties": {
+                "dayOfWeek": {
+                  "type": "string",
+                  "enum": [
+                    "Monday",
+                    "Tuesday",
+                    "Wednesday",
+                    "Thursday",
+                    "Friday",
+                    "Saturday",
+                    "Sunday"
+                  ],
+                  "x-ms-enum": {
+                    "name": "MaintenanceDayOfWeekType",
+                    "modelAsString": false
+                  },
+                  "description": "Day of the week on which the maintenance window will occur."
+                },
+                "timeOfDayUTC": {
+                  "type": "string",
+                  "pattern": "^([0-9]|0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$",
+                  "description": "The time of day (in UTC) to start the maintenance window."
+                }
+              }
+            },
+            "hsm": {
+              "type": "object",
+              "description": "Hydration and archive settings and status",
+              "properties": {
+                "settings": {
+                  "$ref": "#/definitions/AmlFilesystemHsmSettings",
+                  "description": "Specifies HSM settings of the AML file system."
+                },
+                "archiveStatus": {
+                  "type": "array",
+                  "readOnly": true,
+                  "description": "Archive status",
+                  "items": {
+                    "$ref": "#/definitions/AmlFilesystemArchive"
+                  },
+                  "x-ms-identifiers": [
+                    "filesystemPath"
+                  ]
+                }
+              }
+            }
+          },
+          "type": "object",
+          "required": [
+            "filesystemSubnet",
+            "storageCapacityTiB",
+            "maintenanceWindow"
+          ]
+        }
+      }
+    },
+    "AmlFilesystemUpdate": {
```

Comment:
>AmlFilesystemUpdate [](http://example.com/codeflow?start=5&length=19)

I'm not sure what the story is with this being an error instead of a warning

❌ PatchSkuProperty The patch operation body parameter schema should contains property 'sku'.
Location: Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json#L333 RPC-Patch-V1-09

but seems like one possible workaround would be to just declare it as a valid property, and any limitation that you have server side about not yet supporting it, could be handled with a "400 BadRequest not yet implemented support for sku updates"

I guess the dream here is that people might not need to update their api-version/sdk just to use scaling feature once you have it?/cc  @rkmanda

Reply:
> > AmlFilesystemUpdate
> 
> I'm not sure what the story is with this being an error instead of a warning
> 
> ❌ PatchSkuProperty The patch operation body parameter schema should contains property 'sku'. Location: Microsoft.StorageCache/preview/2023-03-01-preview/amlfilesystem.json#L333 RPC-Patch-V1-09
> 
> but seems like one possible workaround would be to just declare it as a valid property, and any limitation that you have server side about not yet supporting it, could be handled with a "400 BadRequest not yet implemented support for sku updates"
> 
> I guess the dream here is that people might not need to update their api-version/sdk just to use scaling feature once you have it?/cc @rkmanda

Thanks.  I asked the API reviewer from devdiv about that check and why I would add fields that we can't modify.  If it changes in the future a new api version could add sku so that it can be modified.